### PR TITLE
"Ancient Artifact" Cargo Crate

### PR DESCRIPTION
**Pull request by Oricana-16**
# About The Pull Request

A new contraband crate you can buy for 30k credits, the Ancient Artifact crates. When you buy it you get a random artifact Nanotrasen decided to sell off

Some of the artifacts include

Daemon Mask: A sentient mask that can possess a person who wears it for 1 minute, and a 2 minute cooldown afterwards. When they possess someone, they get picked up from any stuns and are immune to crits for as long as possession lasts. While possessing the person they can teleport to and stun someone for 3 seconds. While in mask mode they can use Fear on a 1 minute cooldown, which stops people nearby from moving for a few seconds.

Busted Invisibility Matrix: Makes the user nearly invisible for a random time between 5-15 seconds, and goes on a 30 second cooldown afterwards.

Damage Thief: You can take all of a damage type from another living being. You can't take tox or oxy damage if youre immune to it (like an ipc would be)

Organics Smuggling Bag: If you attack people with it, after 5 seconds they go in the bag. If you use the bag while people are inside it, everyone inside falls out. It can hold up to five people.

Soul Binding Contract: A contract that you use on a corpse, after 30 seconds it revives and heals them, but binds your souls together so if either of you dies, so does the other.

## Why It's Good For The Game

This was mostly just for me to learn a bunch of dm quicker and make cool items that I thought of. I think it's good for the game cause more fun items are fun, but it also just makes sense if anyone doesn't want this in the game.|

If this does get accepted and I do it again, I think the max I'll limit myself to is either 10 or 20 total items. 30 total in a worst case scenario where I just keep making more item PRs.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

image

</details>

## Changelog

:cl:
add: Adds Ancient Artifact contraband crate
add: Adds Daemon Mask, Busted Invisibility Matrix, Damage Thief, Organics Smuggling Bag, and Soul Binding Contract to the Ancient Artifact crate.
/:cl:

### CODE REVIEW

**Comment by Oricana-16** [comment]
I started this branch to help me learn DM better and to make cool item ideas I liked, but I'm starting to really like some of these ideas, So I wanted to see if I could get some of the items in in some way (albeit after a bunch of balancing). 

I'm not Too sure about the cargo crate idea, I'd need other's opinions on that. maybe It'd be better to put them in lavaland chests or something.

**Comment by Oricana-16** [comment]
grey line now green line 😱

**Comment by ilikesaying** [comment]
so with the soul bind contract, what happens if one of the souls is revived? does the other soul attempt a revive?

**Comment by Oricana-16** [comment]
No it doesn't, only the one gets revived, though I am considering making it dust them or making them unable to be revived on death to fit the soul binding part more.

File: monkestation/code/game/objects/items/ancient_artifacts.dm [hunk 1]
```diff
@@ -0,0 +1,455 @@
+//Daemon Mask
+/obj/item/clothing/mask/daemon_mask
```
Comment (nednaZ) [inline monkestation/code/game/objects/items/ancient_artifacts.dm:2]: Overall opinion of this one is that it's far too powerful for an easy cargo purchase.
I'd make it a Curator restricted TC item.

File: monkestation/code/game/objects/items/ancient_artifacts.dm [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+		to_chat(user, "<span class='warning'>the [src] isn't ready yet!</span>")
+		return
+
+
+	user.visible_message("<span class='warning'>[user.name] starts to turn transparent!</span>", \
+						"<span class='notice'>Your skin turns transparent.</span>")
+
+	animate(user, alpha = 25,time = 1 SECONDS)
+
+	var/invisibility_time = rand(15 SECONDS, 100 SECONDS)
```
Comment (nednaZ) [inline monkestation/code/game/objects/items/ancient_artifacts.dm:298]: Far too long.
5 to 15 seconds at most.

File: monkestation/code/game/objects/items/ancient_artifacts.dm [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+
+	user.apply_damage(damage,damage_type)
+	if(damage >= 1)
+		user.visible_message("<span class='notice'>[target]'s wounds disappear, as [user] looks more damaged.</span>", \
+							"<span class='notice'>[target] starts to look better, while you feel a little more hurt.</span>")
+	else
+		to_chat(user,"<span class='notice'>Nothing happened, [target] must not've been hurt.</span>")
+
+//Organics Smuggling Bag
+/obj/item/organics_smuggling_bag
```
Comment (nednaZ) [inline monkestation/code/game/objects/items/ancient_artifacts.dm:381]: This one needs a "Resist" interaction to escape.

File: monkestation/code/game/objects/items/ancient_artifacts.dm [hunk 4]
```diff
@@ -0,0 +1,455 @@
...
+
+//Soul Binding Contract
+/obj/item/soul_link_contract
+	name = "soul binding contract"
+	desc = "A contract from Hell itself. If you get a corpse to sign it, it brings them back from beyond the grave and binds your souls together."
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "paper_onfire"
+	item_state = "paper"
+
+/obj/item/soul_link_contract/afterattack(mob/living/target, mob/living/user)
```
Comment (nednaZ) [inline monkestation/code/game/objects/items/ancient_artifacts.dm:432]: This one needs to check for DNR and also offer a chance for the ghost to say no.

**Review by nednaZ — CHANGES_REQUESTED** [review]
My biggest issue is that these can just be purchased directly.
I'd rather have these take some form of effort OR random chance.
If there was a random chance that this crate would mysteriously show up through a rare event, that could be good.

The four items that weren't too OP should also have their own subtype, so you can just have it select from that subtype when adding them.

File: monkestation/code/game/objects/items/ancient_artifacts.dm (hunk 1 above)
Comment (Oricana-16) [reply monkestation/code/game/objects/items/ancient_artifacts.dm:2]: Do you care if it's in this PR or should it be a separate one

File: monkestation/code/game/objects/items/ancient_artifacts.dm [hunk 5]
```diff
@@ -0,0 +1,455 @@
...
+		return
+	if(!(GLOB.ghost_role_flags & GHOSTROLE_STATION_SENTIENCE))
+		to_chat(user, "<span class='notice'>The mask doesn't react, it must be broken!</span>")
+		return
+
+	to_chat(user, "[src] starts glowing...")
+
+	possessed = TRUE
+
+	var/list/mob/dead/observer/candidates = pollGhostCandidates("Do you want to play as the spirit of the daemon mask?", ROLE_PAI, null, FALSE, 100, POLL_IGNORE_POSSESSED_BLADE)
```
[the remark this answers is not in the document]
Comment (Oricana-16) [reply monkestation/code/game/objects/items/ancient_artifacts.dm:37]: The mask isn't really bound to the user is the thing, so it's not Super important who's summoning them. I can still put who's using mask though if you want

[the remark this answers is not in the document]
Comment (Oricana-16) [reply monkestation/code/game/objects/items/ancient_artifacts.dm:37]: The mask isn't bound to the user so it doesn't matter as much who is summoning them. I can still put who it is though if you think it should be like that.

**Comment by Oricana-16** [comment]
Should the random event be something like cargo gets sent the crate like how I think some other events do?

**Comment by Oricana-16** [comment]
After my other traitor items pr is done, I'm gonna make the mask curator tot item pr. also gonna close this one cause it already feels very iffy, and moving the mask out pushes it too far into the area of me not wanting it.

**Pull request closed without merging**